lbuild/node.py

```diff
@@ -286,19 +309,17 @@ def add_dependencies(self, *dependencies):
         """
         self._dependency_module_names += dependencies
 
-    def add_option(self, option):
-        """
-        Define new option for this module.
-
-        The module options only influence the build process but not the
-        selection and dependencies of modules.
-        """
-        if option.name in [child.name for child in self.children]:
-            raise LbuildException("Option name '{}' is already defined".format(option.name))
-        option._repository = self._repository
-        option.parent = self
-        option.add_dependencies(self.fullname)
-        option._fullname = self.fullname + ":" + option.name
+    def add_child(self, node):
```

Comment:
Here you introduce "child" without ever defining what a child is in this context. This becomes confusing for the user when this naming is presented in the exception error text. The user only defines options, queries, etc. but not children.

Reply:
Good point, I'll format with `child.__class__.__name__` instead.